### PR TITLE
docs(hub): add debug section

### PR DESCRIPTION
**Pull request by mapleeit**

### CODE REVIEW

**Comment by JoanFM** [comment]
@samsja I am not sure if it collides or complements your idea for the ticket #4406

**Comment by alexcg1** [comment]
@mapleeit can you run this through grammarly? I'm seeing grammar issues

**Comment by alexcg1** [comment]
Also, be careful on this stuff:

## Capitalization

Please capitalize first letters of the following (when used to reference project or component):

- Flow
- Executor
- Document
- Jina
- Jina Hub
- Python
- Docker
- Kubernetes

### Note!

This only applies **outside** a code or CLI context. If you're describing Python, YAML, any programming language, or command line input/output, please use the capitalization you would use **in that context**.

##### Right

```python
from jina.flow import Flow
```

##### Wrong

```python
from Jina.Flow import Flow
```

---

Made with [GitHub saved replies](https://github.com/settings/replies)

**Comment by JohannesMessner** [comment]
Is it possible to describe in text form what the .gif shows?
Is the message here that the user can debug the Executor like any other Python script that they wrote themselves? If so, I think we should just say that, then they immediately know what to do.
If not, what are the extra steps?

My belief is that gifs/videos should be sparingly used, and be in addition to a written explanation, not instead of.
Would be great if you could add a few lines!

**Comment by samsja** [comment]
> @samsja I am not sure if it collides or complements your idea for the ticket #4406

Nop, it does not collide. 

I am not sure to see the point of this PR (I might be missing something). When do users will want to debug `Executer` like this ?

I would say that it is almost never the recommended way to debug `Executor`. The code in `.jina` is only managed by jina and should never be touched by user right ?

If I want to a modification on an executor. I should clone the repo of the executor, and work from it.

**Comment by JohannesMessner** [comment]
> When do users will want to debug `Executer` like this ?

I think it could be useful when getting a hub executor to work in your Flow. So you don't want to change the Executor itself, let's say it works as intended, but you still want to look inside to see how exactly it behaves.

**Comment by samsja** [comment]
> > When do users will want to debug `Executer` like this ?
> 
> I think it could be useful when getting a hub executor to work in your Flow. So you don't want to change the Executor itself, let's say it works as intended, but you still want to look inside to see how exactly it behaves.

For me it is too hacky to be in ours docs. People will get confuse.

**Comment by alexcg1** [comment]
@JohannesMessner that sounds good. @Shubhamsaboo handles this kind of thing.

Or we could add it to FAQ "how do I debug Executors?"

**Comment by mapleeit** [comment]
> Or we could add it to FAQ "how do I debug Executors?"

Where is the FAQ? If there is, I'm happy to add it there since the content is pretty short.

**Comment by mapleeit** [comment]
> My belief is that gifs/videos should be sparingly used, and be in addition to a written explanation, not instead of.
Would be great if you could add a few lines!

@JohannesMessner Agree on what you say. I can just remove it since it's just show people how to do it intuitively, no additional information.

**Review by samsja — COMMENTED** [review]
Then, as @alexcg1 proposed, it is better to move to it the FAQ

File: docs/fundamentals/executor/hub/debug-executor.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Debug Executor
+
+You need to debug Jina Hub Executor for some reason. It's not so easy to debug inside the Docker container, but Jina Hub Executor provides a way to use source code. It's easy to debug using source code. 
+
+I'll take debugging [Hello Executor](https://hub.jina.ai/executor/9o9yjq1q) as an example. Tell you step by step how to do that.
```
Comment (CatStark) [inline docs/fundamentals/executor/hub/debug-executor.md:5]:
```suggestion
I'll use [Hello Executor](https://hub.jina.ai/executor/9o9yjq1q) as an example for debugging and show you step by step how to do that. 
```

**Review by CatStark — COMMENTED** [review]
Style

**Comment by mapleeit** [comment]
@samsja @alexcg1  Can't find the FAQ in our documentation. So I move it to the `How-to` part.

**Comment by mapleeit** [comment]
@samsja @JohannesMessner Do you have further question about this?

**Comment by samsja** [comment]
> @samsja @JohannesMessner Do you have further question about this?

lgtm :)

**Review by samsja — DISMISSED** [review]

**Comment by mapleeit** [comment]
> can u please open the PR from a branch from jina fork and not from ur fork? This will trigger all proper CI

Open another PR https://github.com/jina-ai/jina/pull/4425

**Pull request closed without merging**